lib/provider-manager.coffee

```diff
@@ -93,7 +116,10 @@ class ProviderManager
   isValidProvider: (provider, apiVersion) ->
     # TODO API: Check based on the apiVersion
     if semver.satisfies(apiVersion, '>=2.0.0')
-      provider? and isFunction(provider.getSuggestions) and isString(provider.selector) and !!provider.selector.length
+      provider? and
+      isFunction(provider.getSuggestions) and
+      ((isString(provider.selector) and !!provider.selector.length) or
```

Comment:
Could you use the `grim` library to post a deprecation message when people use the old field name?